src/state/slices/marketDataSlice/selectors.ts

```diff
@@ -32,13 +34,48 @@ export const selectPriceHistoryByAssetTimeframe = createSelector(
   (priceHistory, assetId, timeframe) => priceHistory[timeframe][assetId] ?? []
 )
 
+const selectPriceHistoryQueryByAssetTimeframe = createSelector(
+  (_state: ReduxState, assetId: CAIP19, _timeframe: HistoryTimeframe) => assetId,
+  (_state: ReduxState, _assetId: CAIP19, timeframe: HistoryTimeframe) => timeframe,
+  (assetId, timeframe) =>
+    marketApi.endpoints.findPriceHistoryByCaip19.select({ assetId, timeframe })
+)
+
+export const selectPriceHistoryLoadingByAssetTimeframe = createSelector(
+  (state: ReduxState) => state,
+  selectPriceHistoryQueryByAssetTimeframe,
+  (state: ReduxState, querySelector) => querySelector(state).isLoading
+)
+
+export const selectPriceHistoryUnavailableByAssetTimeframe = createSelector(
+  (state: ReduxState, _assetId: CAIP19, _timeframe: HistoryTimeframe) => state,
+  selectPriceHistoryQueryByAssetTimeframe,
+  (state: ReduxState, querySelector) =>
+    querySelector(state).isSuccess && querySelector(state).data?.length === 0
+)
+
 export const selectPriceHistoriesLoadingByAssetTimeframe = createSelector(
-  selectPriceHistory,
+  (state: ReduxState) => state,
+  (_state: ReduxState, assetIds: CAIP19[], _timeframe: HistoryTimeframe) => assetIds,
+  (_state: ReduxState, _assetIds: CAIP19[], timeframe: HistoryTimeframe) => timeframe,
```

Comment:
`selectAssetIds` and `selectTimeFrame` ? We use this pattern a lot and it makes it easy to read that an argument is being selected.